ts/store/reducers/__tests__/featureFlagWithMinAppVersionStatus.test.ts

```diff
@@ -0,0 +1,529 @@
+      min_app_version: { ios: "3.0.0.0", android: "3.0.0.0" }
+    });
+    expect(
+      isPropertyWithMinAppVersionEnabled({
+        backendStatus: store,
+        mainLocalFlag: false,
+        configPropertyName: "nativeLogin"
+      })
+    ).toBeFalsy();
+  });
+
+  it("should return (nativelogin) disabled --- min version undefined & local flag disabled", () => {
+    const store = customConfigNativeLogin({});
+    expect(
+      isPropertyWithMinAppVersionEnabled({
+        backendStatus: store,
+        mainLocalFlag: false,
+        configPropertyName: "nativeLogin"
+      })
+    ).toBeFalsy();
+  });
+
+  it("should return (nativelogin) disabled --- min version undefined & local flag enabled", () => {
+    const store = customConfigNativeLogin({});
+    expect(
+      isPropertyWithMinAppVersionEnabled({
+        backendStatus: store,
+        mainLocalFlag: true,
+        configPropertyName: "nativeLogin"
+      })
+    ).toBeFalsy();
+  });
+
+  it("should return (nativelogin) disabled --- nativelogin undefined & local flag enabled", () => {
+    const store = O.some({
+      config: {}
+    }) as unknown as O.Option<BackendStatus>;
+    expect(
+      isPropertyWithMinAppVersionEnabled({
+        backendStatus: store,
+        mainLocalFlag: true,
+        configPropertyName: "nativeLogin"
+      })
+    ).toBeFalsy();
+  });
+
+  it("should return (nativelogin) disabled --- nativelogin undefined & local flag disabled", () => {
+    const store = O.some({
+      config: {}
+    }) as unknown as O.Option<BackendStatus>;
+    expect(
+      isPropertyWithMinAppVersionEnabled({
+        backendStatus: store,
+        mainLocalFlag: false,
+        configPropertyName: "nativeLogin"
+      })
+    ).toBeFalsy();
+  });
+});
+
+describe("backend service Feature Flag with min app version --- with optional config", () => {
+  jest.spyOn(DeviceInfo, "getVersion").mockReturnValue("2.0.0.0");
+  jest.spyOn(DeviceInfo, "getReadableVersion").mockReturnValue("2.0.0.0");
+
+  const customConfigFastLogin = (
+    fastLoginConfig: object
+  ): O.Option<BackendStatus> =>
+    O.some({
+      config: {
+        fastLogin: fastLoginConfig
+      }
+    }) as unknown as O.Option<BackendStatus>;
+
+  it("should return (fastlogin) enabled --- eq min version & local flag enabled & optional config undefined", () => {
+    const store = customConfigFastLogin({
+      min_app_version: { ios: "2.0.0.0", android: "2.0.0.0" }
+    });
+    expect(
+      isPropertyWithMinAppVersionEnabled({
+        backendStatus: store,
+        mainLocalFlag: true,
+        configPropertyName: "fastLogin"
+      })
+    ).toBeTruthy();
+  });
+
+  it("should return (fastlogin) enabled --- eq min version & local flag enabled & optional config > min version", () => {
+    const store = customConfigFastLogin({
+      min_app_version: { ios: "2.0.0.0", android: "2.0.0.0" },
+      opt_in: {
+        min_app_version: { ios: "3.0.0.0", android: "3.0.0.0" }
+      }
```

Comment:
Do we need this here?

Reply:
Just to test what happens if we have an nested configuration in the structure, but we don't want its state (enabled or not)